abb_hardware_interface/include/abb_hardware_interface/rws_service_provider_ros.hpp

```diff
@@ -0,0 +1,230 @@
+
+  bool set_rapid_routine(const abb_rapid_sm_addin_msgs::srv::SetRAPIDRoutine::Request::SharedPtr req,
+                         abb_rapid_sm_addin_msgs::srv::SetRAPIDRoutine::Response::SharedPtr res);
+
+  bool set_sg_command(const abb_rapid_sm_addin_msgs::srv::SetSGCommand::Request::SharedPtr req,
+                      abb_rapid_sm_addin_msgs::srv::SetSGCommand::Response::SharedPtr res);
+
+  bool start_egm_joint(const abb_robot_msgs::srv::TriggerWithResultCode::Request::SharedPtr req,
+                       abb_robot_msgs::srv::TriggerWithResultCode::Response::SharedPtr res);
+
+  bool start_egm_pose(const abb_robot_msgs::srv::TriggerWithResultCode::Request::SharedPtr req,
+                      abb_robot_msgs::srv::TriggerWithResultCode::Response::SharedPtr res);
+
+  bool start_egm_stream(const abb_robot_msgs::srv::TriggerWithResultCode::Request::SharedPtr req,
+                        abb_robot_msgs::srv::TriggerWithResultCode::Response::SharedPtr res);
+
+  bool stop_egm(const abb_robot_msgs::srv::TriggerWithResultCode::Request::SharedPtr req,
+                abb_robot_msgs::srv::TriggerWithResultCode::Response::SharedPtr res);
+
+  bool stop_egm_stream(const abb_robot_msgs::srv::TriggerWithResultCode::Request::SharedPtr req,
+                       abb_robot_msgs::srv::TriggerWithResultCode::Response::SharedPtr res);
+
+  bool verify_auto_mode(uint16_t& result_code, std::string& message);
+
+  bool verify_argument_filename(const std::string& filename, uint16_t& result_code, std::string& message);
+
+  bool verify_argument_rapid_symbol_path(const abb_robot_msgs::msg::RAPIDSymbolPath& path, uint16_t& result_code,
+                                         std::string& message);
+
+  bool verify_argument_rapid_task(const std::string& task, uint16_t& result_code, std::string& message);
+
+  bool verify_argument_signal(const std::string& signal, uint16_t& result_code, std::string& message);
+
+  bool verify_motors_off(uint16_t& result_code, std::string& message);
+
+  bool verify_motors_on(uint16_t& result_code, std::string& message);
+
+  bool verify_sm_addin_runtime_states(uint16_t& result_code, std::string& message);
+
+  bool verify_sm_addin_task_exist(const std::string& task, uint16_t& result_code, std::string& message);
+
+  bool verify_sm_addin_task_initialized(const std::string& task, uint16_t& result_code, std::string& message);
+
+  bool verify_rapid_running(uint16_t& result_code, std::string& message);
+
+  bool verify_rapid_stopped(uint16_t& result_code, std::string& message);
+
+  bool verify_rws_manager_ready(uint16_t& result_code, std::string& message);
+
+  rclcpp::Subscription<abb_robot_msgs::msg::SystemState>::SharedPtr system_state_sub_;
+  rclcpp::Subscription<abb_rapid_sm_addin_msgs::msg::RuntimeState>::SharedPtr runtime_state_sub_;
+
+  std::vector<rclcpp::ServiceBase::WeakPtr> core_services_;
+  std::vector<rclcpp::ServiceBase::WeakPtr> sm_services_;
```

Comment:
These vectors to me seem to have ownership over the services, so I'd argue they should be SharedPtr.